osaca/parser/parser_x86intel.py

```diff
@@ -203,6 +262,33 @@ def process_register(self, operand):
             name=operand["name"],
         )
 
+    def process_memory_address(self, memory_address):
+        """Post-process memory address operand"""
+        # TODO: Use the ptr type.
+        ptr_type = memory_address["ptr_type"]
+        register_expression = memory_address["register_expression"]
+        displacement = register_expression.get(
+            "displacement1",
+            register_expression.get("displacement2", None)
+        )
+        base = register_expression.get("base", None)
+        index = register_expression.get("index", None)
+        scale = 1 if "scale" not in register_expression else int(register_expression["scale"], 0)
+        displacementOp = None
+        baseOp = None
+        indexOp = None
+        if displacement:
+            displacementOp = self.process_immediate(displacement["immediate"])
+        if base:
+            baseOp = RegisterOperand(name=base["name"])
+        if index:
+            indexOp = RegisterOperand(name=index["name"])
```

Comment:
Alternatively, with conditional expressions:
`displacementOp = self.process_immediate(displacement["immediate"]) if displacement else None`
etc.

Reply:
Also, you can use `displacement.immediate` instead of `displacement["immediate"]`, passim mutatis mutandis. See https://pyparsing-docs.readthedocs.io/en/latest/pyparsing.html#pyparsing.ParseResults.